utils/extra.py

```diff
@@ -93,28 +95,28 @@ async def post(bot, code):
         "embed_colour": "#FFFFFF",
     }
 
-    response = await bot.session.post(
+    async with await bot.session.post(
         "https://api.senarc.org/paste",
         json=paste_body,
         headers={"accept": "application/json", "Content-Type": "application/json"},
-    )
-    response = await response.json()
-    return response.get("url")
+    ) as response:
+        data = await response.json()
+        return data.get("url")
 
 
-async def get_paste(bot, paste_id):
-    response = await bot.session.get(
+async def get_paste(bot, paste_id) -> typing.Optional[str]:
+    async with bot.session.get(
         f"https://api.senarc.org/bin/{paste_id}", headers={"accept": "application/json", "headless": "true"}
-    )
-    response = await response.json()
-    return response.get("content")
+    ) as response:
+        data = await response.json()
+        return data.get("content")
 
 
-def random_history(data, number):
+def random_history(data, number) -> list[typing.Any]:
     return random.sample(data, number)
 
 
-def groupby(iterable: list, number: int):
+def groupby(iterable: list, number: int) -> list[int]:
```

Comment:
consider
```py
def groupby(iterable: list[T], number: int) -> list[list[T]]:
```

where T is a generic TypeVar

Reply:
Done. (except as for code body change, i'll see it later)